src/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets/SocketConnection.cs

```diff
@@ -207,6 +204,8 @@ private void SetupSendBuffers(ReadableBuffer buffer)
                         _output.Advance(buffer.End);
                     }
                 }
+
+                _socket.Shutdown(SocketShutdown.Send);
```

Comment:
Do managed sockets send a FIN implicitly like libuv if you close the socket without calling Shutdown?